### DISCUSSION TITLE
Insert data with pgp_sym_encrypt not working with function

### DISCUSSION THREAD

**Discussion by jsonpreet** [supabase/supabase#20936] [Questions]
Hi,

I am trying to insert data from my mobile app into supabase data using sql function.

```
CREATE OR REPLACE FUNCTION public.insertProjectByUserId(title text, description text, user_id uuid, key text) RETURNS projects LANGUAGE plpgsql AS $$
declare
  enc_title text;
  enc_description text;
BEGIN
  enc_title := pgp_sym_encrypt(title, key);
  enc_description := pgp_sym_encrypt(description, key);
  INSERT INTO public.projects (title, description, user_id) VALUES (enc_title, enc_description, user_id);
  SELECT  new.id, pgp_sym_decrypt(new.title::bytea, key), pgp_sym_decrypt(new.description::bytea, key), user_id, new.created_at;
  return new;
END;
$$ ;
```

When I am submitting data from App using function it store data in plain text not encrypted 
I used query with supabase-js

`await supabase.rpc('insertProjectByUserId', { title, description, user_id, key })`

When I tried single insert query from sql editor and it's working.

`INSERT INTO public.projects (title, description, user_id) VALUES (pgp_sym_encrypt('text', '2c75f1689f45410977ac006394298d0efb4bac8c3964867251616e2cebc7d367'), pgp_sym_encrypt('description', '2c75f1689f45410977ac006394298d0efb4bac8c3964867251616e2cebc7d367'), '34d024c8-82fb-483d-a059-895f440f10db')`

**Comment by im2x3m4u** [comment]
i get the same issu here. is there any solution or workaround about this?

**Comment by eusthace811** [comment]
Having the same problem, any help?
